### PR TITLE
Use futures sub-crates explicitly

### PR DESCRIPTION
**Pull request by Byron**
This bring initial compile times (using `cargo check`) down from ~22s to ~14s, while reducing the amount of dependencies from 25 to 23.

Please let me know if you would like to see any changes.

**edit**: These amazing build time improvements cannot be reproduced anymore. Probably just a local fluke :/.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -13,7 +13,10 @@ categories = ["asynchronous", "concurrency"]
 readme = "README.md"
 
 [dependencies]
-futures = { version = "0.3.4", default-features = false, features = ["std"] }
+futures-util = { version = "0.3.5", default-features = false, features = ["std"] }
+futures-channel = { version = "0.3.5", default-features = false, features = ["std"] }
+futures-core = { version = "0.3.5", default-features = false, features = ["std"] }
```
Comment (taiki-e) [inline Cargo.toml]: futures-util exports all items defined in futures-core, so we don't need to depend on futures-core directly.

File: Cargo.toml [hunk 2]
```diff
@@ -13,7 +13,10 @@ categories = ["asynchronous", "concurrency"]
 readme = "README.md"
 
 [dependencies]
-futures = { version = "0.3.4", default-features = false, features = ["std"] }
+futures-util = { version = "0.3.5", default-features = false, features = ["std"] }
```
Comment (taiki-e) [inline Cargo.toml]:
```suggestion
futures-util = { version = "0.3.5", default-features = false, features = ["std", "sink"] }
```

In futures-util, [sink is not enabled by default](https://github.com/rust-lang/futures-rs/blob/master/futures-util/Cargo.toml#L22).

([CI passes because of the feature flag of the dev-dependencies leaks to normal dependencies.](https://github.com/rust-lang/cargo/issues/4866))

**Comment by Byron** [comment]
@taiki-e  Thanks for the review! Do you see a way to make CI fail? Or asked differently: How did you detect the missing 'sink' feature? Maybe CI can be tuned to do the same.

**Comment by taiki-e** [comment]
> How did you detect the missing 'sink' feature? Maybe CI can be tuned to do the same.

By running `cargo check -Z features=dev_dep`, that can be detected (require nightly).

File: .github/workflows/build-and-test.yaml [hunk 3]
```diff
@@ -33,11 +33,19 @@ jobs:
             override: true
 
       - name: Run cargo check
+        if: "!startsWith(matrix.rust, 'nightly')"
```
Comment (ghost) [inline .github/workflows/build-and-test.yaml]: It would be good to check everything (benches, bins, examples, tests) on nightly anyways.
```suggestion
```

Comment (Byron) [reply .github/workflows/build-and-test.yaml]: It's odd, apparently I don't fully understand the `-Z features=dev_dep`. Let me try again to achieve something like this:

> By running cargo check -Z features=dev_dep, that can be detected (require nightly).

Maybe it's enough to literally run it like that, without additional flags, in addition to what's always run.
Commit is coming up.

**Comment by Byron** [comment]
Oh, interesting, now it does indeed fail the CI. Do you have suggestions on how to fix it? If not, I will install nightly and see what I can do.

**Comment by taiki-e** [comment]
Oh, futures-channel also needs to enable`sink` feature.
